src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -28,17 +28,19 @@
                                        help="The parent path (Ex: 'resA/myA/resB/myB')")
 _PROVIDER_HELP_TEXT = 'the resource namespace, aka \'provider\''
 register_cli_argument('resource', 'no_wait', no_wait_type)
-register_cli_argument('resource', 'resource_name', resource_name_type)
+register_cli_argument('resource', 'resource_id', ignore_type)
```

Comment:
I would suggest call out in the history.pst file that we are hiding this field. In code emit out warnings on such usages, then after one release, let us get rid of it (open a track issue for this)